File: dotnet/PoeLogsParser/Models/Entries/TradeLogEntry.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Text.RegularExpressions;
using PoeLogsParser.Enums;

namespace PoeLogsParser.Models
{
    public class TradeLogEntry : LogEntry
    {
        public string Player { get; set; }
        public Item Item { get; set; }
        public string League { get; set; }
        public Price Price { get; set; }
        public StashLocation Location { get; set; }
        public string Notes { get; set; }

        public TradeLogEntry(LogEntry entry) : base(entry.Raw, entry.Time, entry.Types, entry.Tag)
        {
        }
        
       
    }
}

File: ts/src/models/entries/TradeLogEntry.ts
import { Item } from "../Item";
import { Price } from "../Price";
import { StashLocation } from "../StashLocation";
import { LogEntry } from "./LogEntry";

export class TradeLogEntry extends LogEntry {
	public player!: String;
	public item!: Item;
	public league!: String;
	public price!: Price;
	public location!: StashLocation;

	constructor(entry: LogEntry) {
		super(entry.raw as string, entry.time, entry.types, entry.tag);
	}
}